Spigot-API-Patches/0130-Add-source-block-to-BlockPhysicsEvent.patch

```diff
@@ -0,0 +1,54 @@
+From 82beae7d9ee7d5e6b56cf7e8ae8fe072087b318f Mon Sep 17 00:00:00 2001
+From: Sotr <i@omc.hk>
+Date: Wed, 22 Aug 2018 17:20:37 +0800
+Subject: [PATCH] Add source block to BlockPhysicsEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/block/BlockPhysicsEvent.java b/src/main/java/org/bukkit/event/block/BlockPhysicsEvent.java
+index 01a545b..41b135f 100644
+--- a/src/main/java/org/bukkit/event/block/BlockPhysicsEvent.java
++++ b/src/main/java/org/bukkit/event/block/BlockPhysicsEvent.java
+@@ -11,6 +11,7 @@ import org.bukkit.event.HandlerList;
+ public class BlockPhysicsEvent extends BlockEvent implements Cancellable {
+     private static final HandlerList handlers = new HandlerList();
+     private final int changed;
++    private final Block source; // Paper
+     private boolean cancel = false;
+ 
+     /**
+@@ -21,9 +22,32 @@ public class BlockPhysicsEvent extends BlockEvent implements Cancellable {
+      */
+     @Deprecated
+     public BlockPhysicsEvent(final Block block, final int changed) {
++        this(block, changed, null); // Paper
++    }
++
++    // Paper start
++    /**
++     *
++     * @deprecated Magic value
++     * @param block the block involved in this event
++     * @param source the changed block
++     */
++    @Deprecated
++    public BlockPhysicsEvent(final Block block, final int changed, @javax.annotation.Nullable final Block source) {
+         super(block);
+         this.changed = changed;
++        this.source = source == null ? block : source;
++    }
++
++    /**
++     * Gets the block that changed, causing this event
++     *
++     * @return Changed block
++     */
++    public Block getChangedBlock() {
```

Comment:
Let's rename this to as the patch name states, getSourceBlock. getChangedBlock feels wrong, 

additionally, look up block here with world.getBlockAt, so the getblock is only called if someone uses this API.